### ISSUE TITLE
CLI export option for PDF with plain background

### ISSUE THREAD

**Issue by schaetzc** [xournalpp/xournalpp#1464]
### Is your feature request related to a problem?
I can export journals to PDFs without the paper background (e.g. lined, dotted, …) using File > Export As ... > PDF with plain background. However, I have to open the GUI to do so. I would like to do it in a script. At the moment PDFs with backgrounds can be exported from the command line, but I haven't found a way to export PDFs without backgrounds.  

### Describe the solution you'd like
Add new command line options to export PDFs without backgrounds. It would make sense to implement this as an additional switch as there are also other export formats such as PNG and SVG which should have the same option. Ideally, we would have something like the following, although that would go beyond my feature request by a bit.

```
-p, --create-pdf         PDF output filename
-i, --create-img         Image output filename (.png / .svg)
-P, --plain-background   When exporting, set the page background to be plain
-T, --transparent-paper  When exporting, set the paper color to be transparent
```
Then we could do `xournalpp -P -p out.pdf in.xopp` or `xournalpp -Pp out.pdf in.xopp` to export a PDF with plain page backgrounds as in the GUI.

Please note that `-P` and `-T` are orthogonal. With `-T` the pages still have their lined, ruled, or dotted background as specified in the `.xoff` file, but the paper color is transparent (a good explanation can be found in [this feature request of rmano](https://github.com/xournalpp/xournalpp/issues/490#issuecomment-445562967)).

Here is a translation between the options offered by the GUI and the CLI
```
-p   out.pdf = PDF files  
-PTp out.pdf = PDF with plain background  
-i   out.png = PNG graphics  
-PTi out.png = PNG with transparent background  
-i   out.svg = SVG graphics  
-PTi out.svg = SVG with transparent background
```
In addition there are new options currently not offered by the GUI:  
`-Tp out.pdf`, `-Pp out.pdf`, `-Ti out.png/svg`, `-Pi out.png/svg`

Ideally, the export options would also respect the already existing `-n, --page=N` option, so that individual pages can be exported. Support for range expressions as known from printers (for instance `--page=1,3,5-8`) would be even better.

### Describe alternatives you've considered

**Minimal Alternative:**
I thought I better suggest the ideal solution. However, at the moment and for my needs, a single additional export option would be sufficient. The new export option can *not* be combined and acts like "PDF with plain background" in the GUI's export dialoge.
```
-p, --create-pdf
-P, --create-pdf-with-plain-background
-i, --create-img
```

**Maximal Alternative:**
We could also add more options than proposed in the first solution. For instance, instead of setting the page background to "plain" we could allow setting the page background to anything, even for individual pages. Maybe something like `--background=1:plain,3-5:ruled,9:pdf:/path/to/background.pdf:7`. Same goes for the page color.  However, I think at this point it's time to use the GUI as these options are on the same level as options to draw something – even if there was a demand for them, they should be implemented as an API allowing to create/edit xopp files rather than export options.

**Comment by Paumanok** [comment]
This is a feature I've been wanting for a couple years. I have about 5 years worth of school notes from daily classes that I'd like to batch export to pdf and combine into a single PDF for easier reference. The latter could be done elsewhere but the ability to script the PDF export function instead of doing it one by one would be extremely useful, especially as I don't want to accidentally edit files I want to quickly reference.

**Comment by lukts30** [comment]
Even the proposed minimum feature would greatly improve my productivity.

I keep my notes in Markdown and track them with git. I would like to able to embed my handwritten notes like normal images.

```
![](image.xopp.png)
```

Then I could run grep to get a list and invoke export on the referenced files.

**Comment by rolandlo** [comment]
This (at least the minimum version) has been solved with PR #2449. So I will close this issue.
